examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -388,34 +424,37 @@ RegularGridDecomposition::categorizeParticlesIntoLeftAndRightNeighbor(const std:
       leftNeighborParticles.push_back(particle);
 
       // if the particle is outside the global box move it to the other side (periodic boundary)
-      if (_localBoxMin[direction] == _globalBoxMin[direction]) {
+      if (isNear(_localBoxMin[direction], _globalBoxMin[direction])) {
         // TODO: check if this failsafe is really reasonable.
         //  It should only trigger if a particle's position was already inside the box?
         const auto periodicPosition = position[direction] + globalBoxLength[direction];
         const auto justInsideOfBox = std::nextafter(_globalBoxMax[direction], _globalBoxMin[direction]);
         position[direction] = std::min(justInsideOfBox, periodicPosition);
         leftNeighborParticles.back().setR(position);
       }
-    } else  // if the particle is right of the box
-        if (position[direction] >= _localBoxMax[direction]) {
-      rightNeighborParticles.push_back(particle);
-
-      // if the particle is outside the global box move it to the other side (periodic boundary)
-      if (_localBoxMax[direction] == _globalBoxMax[direction]) {
-        // TODO: check if this failsafe is really reasonable.
-        //  It should only trigger if a particle's position was already inside the box?
-        const auto periodicPosition = position[direction] - globalBoxLength[direction];
-        position[direction] = std::max(_globalBoxMin[direction], periodicPosition);
-        rightNeighborParticles.back().setR(position);
-      }
     } else {
-      uncategorizedParticles.push_back(particle);
+      // if the particle is right of the box
+      if (position[direction] >= _localBoxMax[direction]) {
+        rightNeighborParticles.push_back(particle);
+
+        // if the particle is outside the global box move it to the other side (periodic boundary)
+        if (isNear(_localBoxMax[direction], _globalBoxMax[direction])) {
+          // TODO: check if this failsafe is really reasonable.
+          //  It should only trigger if a particle's position was already inside the box?
+          const auto periodicPosition = position[direction] - globalBoxLength[direction];
+          position[direction] = std::max(_globalBoxMin[direction], periodicPosition);
+          rightNeighborParticles.back().setR(position);
+        }
+      } else {
+        uncategorizedParticles.push_back(particle);
+      }
```

Comment:
Why the change of the if-else structure?
From 
```cpp
if () ...
else if () ...
else ...
```
to 
```cpp
if () ...
else {
  if () ...
  else ...
}
```

Reply:
No real reason, but also the two structures should be equivalent.
[2d81b4e](https://github.com/AutoPas/AutoPas/pull/724/commits/2d81b4eba1f70d2c841e27e450b169122a5a65bc)